File: cmd.cpp
#include <iostream>

using namespace std;

int main(int argc, char const *argv[])
{
 /* code */
 for (int i = 1; i < argc; i++)
 {
  if (argv[i][0] != '-')
  {
   cerr << "Caractère - du paramètre manquant" << endl;
   return -1;
  }
  else
  {
   if (argv[i][2] != 0)
   {
    cerr << "Paramètre invalide, trop long";
    return -1;
   }
   switch (argv[i][1])
   {
   case 'l':
    cout << "Le paramètre l est présent";
    break;
   case 'g':
    cout << "Le paramètre g est présent";
    break;
   case 's':
    cout << "Le premier s est présent";
    break;
   default:
    cerr << "Paramètre invalide" << endl;
    return -1;
   }
  }
 }
 return 0;
}

File: Cmd.java
//import java.util.*;

public class Cmd {
 public static void main(String[] args) {
  for (int i = 0; i < args.length; i++) {
   if (args[i].charAt(0) != '-') {
    System.err.println("Caractère - du paramètre manquant!");
    return;
   } else {
    if (args[i].length() != 2) {
     System.err.println("Paramètre invalide");
     return;
    }
    switch (args[i].charAt(1)) {
     case 'l':
      System.out.println("Le paramètre 'l' est présent");

      break;
     case 'g':
      System.out.println("Le paramètre 'g' est présent");
      break;
     case 's':
      System.out.println("Le paramètre 's' est présent");
      break;

     default:
      System.err.println("Paramètre invalide");
      break;
    }
   }
  }
 }
}
